spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -14,4 +16,38 @@
       delete :logout
     end
   end
+
+  describe 'GET /api/saml/auth' do
+    context 'with LOA3 but the identity is already verified' do
+      it 'does not redirect the user to the IdV quiz' do
+        user = create(:user, :signed_up)
+        generate_saml_response(user, loa3_saml_settings)
+
+        user.last_identity.update(verified: true)
+
+        generate_saml_response(user, loa3_saml_settings)
+
+        expect(response).to_not be_redirect
+      end
+    end
+
+    context 'with LOA3 and the identity is not already verified' do
+      it 'redirects to quiz' do
```

Comment:
Should we stop using the word quiz here?   It seems little misleading.   How about 'IdV URL' or 'verification service'?